File: backend/src/main/java/com/ll/hotel/domain/member/member/service/AuthTokenService.java
package com.ll.hotel.domain.member.member.service;


import com.ll.hotel.global.jwt.dto.GeneratedToken;
import com.ll.hotel.global.jwt.dto.JwtProperties;
import com.ll.hotel.standard.util.Ut;
import io.jsonwebtoken.Claims;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

import java.util.Date;
import java.util.HashMap;
import java.util.Map;

@Service
@Slf4j
@RequiredArgsConstructor
public class AuthTokenService {
    private final JwtProperties jwtProperties;
    private final RefreshTokenService refreshTokenService;

    public GeneratedToken generateToken(String email, String role) {
        String accessToken = genAccessToken(email, role);
        String refreshToken = refreshTokenService.generateRefreshToken(email);

        refreshTokenService.saveTokenInfo(email, refreshToken, accessToken);
        return new GeneratedToken(accessToken, refreshToken);
    }

    String genAccessToken(String email, String role) {
        Map<String, Object> claims = new HashMap<>();
        claims.put("sub", email);
        claims.put("role", role);
        claims.put("type", "access");

        return Ut.jwt.toString(jwtProperties, claims);
    }

    boolean verifyToken(String token) {
        try {
            Claims claims = Ut.jwt.getClaims(jwtProperties, token);
            return claims.getExpiration().after(new Date());
        } catch (Exception e) {
            return false;
        }
    }

    String getEmail(String token) {
        return Ut.jwt.getClaims(jwtProperties, token).getSubject();
    }
}

File: backend/src/main/java/com/ll/hotel/domain/member/member/service/AuthTokenService.kt
package com.ll.hotel.domain.member.member.service

import com.ll.hotel.global.jwt.dto.GeneratedToken
import com.ll.hotel.global.jwt.dto.JwtProperties
import com.ll.hotel.standard.util.Ut
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import java.util.*

@Service
class AuthTokenService(
    private val jwtProperties: JwtProperties,
    private val refreshTokenService: RefreshTokenService
) {
    private val log = LoggerFactory.getLogger(AuthTokenService::class.java)

    fun generateToken(email: String, role: String): GeneratedToken {
        val accessToken = genAccessToken(email, role)
        val refreshToken = refreshTokenService.generateRefreshToken(email)

        refreshTokenService.saveTokenInfo(email, refreshToken, accessToken)
        return GeneratedToken(accessToken, refreshToken)
    }

    fun genAccessToken(email: String, role: String): String {
        val claims: MutableMap<String, Any> = HashMap()
        claims["sub"] = email
        claims["role"] = role
        claims["type"] = "access"

        return Ut.jwt.toString(jwtProperties, claims)
    }

    fun verifyToken(token: String): Boolean {
        return try {
            val claims = Ut.jwt.getClaims(jwtProperties, token)
            claims.expiration.after(Date())
        } catch (e: Exception) {
            false
        }
    }

    fun getEmail(token: String): String {
        return Ut.jwt.getClaims(jwtProperties, token).subject
    }
} 